### PR TITLE
Move to cw721 package

### PR DESCRIPTION
**Pull request by taitruong**
This PR only moves code from contracts to cw721 package - as a preparation for PR #156 (`Major release v0.19: add CollectionInfo, RoyaltyInfo, updatable NFTs for creator, etc.`).

@jhernandezb, @JakeHartnell, all checks passed - except lint checker fails with `use of deprecated variant ...`

I have marked deprecated usages with #[allow(deprecated)], but linter still complaints.
Did the same in launchpad PR, but here it is all fine:
https://github.com/public-awesome/launchpad/pull/668/

### CODE REVIEW

File: contracts/cw2981-royalties/src/msg.rs
Comment (humanalgorithm) [inline contracts/cw2981-royalties/src/msg.rs:135]: Is a goal of this PR to enable functionality of cw2981? We are adding all the messages here.

File: contracts/cw721-expiration/src/execute.rs
Comment (humanalgorithm) [inline contracts/cw721-expiration/src/execute.rs:155]: Why are we changing the name here? What's the difference between approve_all and transfer nft include nft expired?

File: contracts/cw721-expiration/src/msg.rs
Comment (humanalgorithm) [inline contracts/cw721-expiration/src/msg.rs:37]: What is difference between "invalid" and "expired" ?

File: contracts/cw721-expiration/src/query.rs
Comment (humanalgorithm) [inline contracts/cw721-expiration/src/query.rs:210]: How come we add "query" in the name? Before they just say "approval" so I'm not sure if naming convention necessitates that we put "query" in the name or not?

File: contracts/cw721-expiration/src/query.rs
Comment (humanalgorithm) [inline contracts/cw721-expiration/src/query.rs:316]: It seems like this "expired nft" is the bulk of the PR. What exactly is an "expired nft?"

**Comment by humanalgorithm** [comment]
Can you better explain why we have this giant app centered around cw721 expiration? Shouldn't "expiration" kind of just be an option handed to a regular cw721 rather than a giant app?

**Review by humanalgorithm — APPROVED** [review]

File: contracts/cw2981-royalties/src/msg.rs
Comment (taitruong) [reply contracts/cw2981-royalties/src/msg.rs:135]: as discussed, `cw2981` will be removed in next PR #156
Here it just got refactored. Before Cw721QueryMsg was wrapped inside Cw721's QueryMsg. So here it explicitely follows same struct as for cw721. But yeah - will be removed anyways. this contract is useless and not production ready, imo

File: contracts/cw721-expiration/src/execute.rs
Comment (taitruong) [reply contracts/cw721-expiration/src/execute.rs:155]: `transfer_nft_include_nft_expired` has nothing to do with `approve_all`. Diff just shows previous line has now different code.

Also here: most logic has been moved to package. Like `approve_all`. This `cw721-expiration` has some specific logic, where expiration needs to be considered. Like for `ExecuteMsg::TransferNft`, expiration needs to be considered in contract, where as other msgs (e.g. `ApproveAll`) is independent from expiration - hence handled in package.

File: contracts/cw721-expiration/src/msg.rs
Comment (taitruong) [reply contracts/cw721-expiration/src/msg.rs:37]: Prop `include_expired` is in general for cw721-base and covers expiration of approvals. Now cw721-expiration includes expiration of NFTs. Term `invalid` wasnt precised, so it got renamed `include_invalid` -> `include_expired_nft`.

Please note for cw721-expiration: I created this new contract for a specific use case (like issuing 12months subscription using NFT expiring after 12 months).

File: contracts/cw721-expiration/src/query.rs
Comment (taitruong) [reply contracts/cw721-expiration/src/query.rs:210]: again, like above diff here, just show it propery. all general cw721 queries are moved into package. All contract specific logic are kept here.

File: contracts/cw721-expiration/src/query.rs
Comment (taitruong) [reply contracts/cw721-expiration/src/query.rs:316]: Example: you provided subscriptions, like a user can subscribe a domain/node/any service for 12 months. Now you user can buy this service, by buying an NFT which expires after 12 months. So creator can create a platform, where user can approve/have access by this NFT, once it got expired, subscription ends.

**Comment by taitruong** [comment]
> Can you better explain why we have this giant app centered around cw721 expiration? Shouldn't "expiration" kind of just be an option handed to a regular cw721 rather than a giant app?

cw721-expiration is specific - and should not be part for cw721-base. Besides contract is already part of previous version. This PR here just did some refactorings.

File: contracts/cw2981-royalties/src/msg.rs
Comment (shanev) [reply contracts/cw2981-royalties/src/msg.rs:135]: We cannot remove cw2981 as its being used by other chains such as Sei. Ideally any changes are backwards compatible.

File: packages/cw721/src/execute.rs
Comment (taitruong) [inline packages/cw721/src/execute.rs:40]: @shanev look here. Like `Cw721Execute` in cw721 package. It is still an interface/trait. Only difference here is that it provides a default implementation like here in `instantiate()`. This way contract just can implement `Cw721Execute` and in case it custom logic is required, contract can implement its own logic - but in 99% of the cases it can just use default logic.

Comment (taitruong) [reply packages/cw721/src/execute.rs:40]: We did the same for `cw-ics721`, by providing default logic - like many other projects: https://github.com/public-awesome/cw-ics721/blob/main/packages/ics721/src/execute.rs#L36-L48

File: Cargo.toml
Comment (JakeHartnell) [inline Cargo.toml]: Where did this come from? If we're going to use a `git` dependency, can we at least tie it to a specific commit / rev.

File: contracts/cw2981-royalties/src/msg.rs
Comment (JakeHartnell) [inline contracts/cw2981-royalties/src/msg.rs]:
```suggestion
```

File: contracts/cw721-expiration/examples/schema.rs
Comment (JakeHartnell) [inline contracts/cw721-expiration/examples/schema.rs:7]: Nit, should include execute messages in schemas because `ts-codegen` needs them for frontend tooling.

File: packages/cw721/examples/schema.rs
Comment (JakeHartnell) [inline packages/cw721/examples/schema.rs:1]: Not sure if we need a schema for the package.

File: packages/cw721/schema/all_nft_info_response.json
Comment (JakeHartnell) [inline packages/cw721/schema/all_nft_info_response.json:1]: Also don't need all of these JSON files for each response type commited. I vote we get rid of `schema.rs` and these schema files for the `cw721` package.

File: contracts/cw721-expiration/examples/schema.rs
Comment (taitruong) [reply contracts/cw721-expiration/examples/schema.rs:7]: Thx! Missed that. Fixed and added.

File: contracts/cw2981-royalties/src/msg.rs
Comment (taitruong) [reply contracts/cw2981-royalties/src/msg.rs]: This was some test code from me, I removed it.

File: Cargo.toml
Comment (taitruong) [reply Cargo.toml]: Fixed. now using a revision.
`cw-ownable` is a singleton, made a fork so it can used multiple times, Shane and Jorge also says this makes sense. Larry said he's fine with it and wanted to merge (which he hasnt done yet): https://github.com/larry0x/cw-plus-plus/pull/21

File: packages/cw721/examples/schema.rs
Comment (taitruong) [reply packages/cw721/examples/schema.rs:1]: Quick note: there was schemas before the change. it looks like ts-codegen checks them for generating client code.

**Review by shanev — APPROVED** [review]

**Review by JakeHartnell — APPROVED** [review]

**Pull request merged**
